core/sql/executor/ex_transaction.cpp

```diff
@@ -1316,11 +1309,29 @@ short ExTransTcb::work()
           // create a diagsArea and return it to the parent.
           handleErrors(pentry_down, ta->getDiagsArea());
 
+        if (cliGlobals->currContext()->ddlStmtsExecuted())
+          {
+            ComDiagsArea * diagsArea = NULL;
+            ExSqlComp::ReturnStatus cmpStatus = 
+              cliGlobals->currContext()->sendXnMsgToArkcmp
+              (NULL, 0,
+               EXSQLCOMP::DDL_NATABLE_INVALIDATE,
+               diagsArea);
+            if (cmpStatus == ExSqlComp::ERROR)
+              {
+                handleErrors(pentry_down, NULL, 
+                             (ExeErrorCode)(-EXE_CANT_COMMIT_OR_ROLLBACK));
+ 
+                return -1;
+              }
+          }
+        
+        cliGlobals->currContext()->ddlStmtsExecuted() = FALSE;
```

Comment:
The ddlStmtsExecuted flag will be set to TRUE if there is any error on commit or rollback. How can the user process recover from this,